clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -2285,42 +2264,47 @@ class SyclKernelIntHeaderCreator : public SyclKernelFieldHandler {
     return true;
   }
 
-  bool enterStruct(const CXXRecordDecl *, FieldDecl *FD) final {
+  bool enterStruct(const CXXRecordDecl *, FieldDecl *FD, QualType) final {
     ++StructDepth;
+    // TODO: Is this right?! I think this only needs to be incremented when we
```

Comment:
This IS wrong thanks to my refactor... Still working through some issues on it, but this is my last issue as far as I know.